sunpy/net/jsoc/jsoc.py

```diff
@@ -702,18 +690,19 @@ def _make_recordset(self, series, start_time='', end_time='', wavelength='',
         keys = []
         keyword_info = c.keys(series)
         for key, value in keyword.items():
-            if key in keyword_info:
-                keys.append(f"{key}{value['operator']}{value['value']}")
-            else:
+            if key not in keyword_info:
                 raise ValueError(f"Keyword: '{key}' is not supported by series: {series}")
+            # We have to ensure that any values that are strings are quoted
+            other_value = value['value'] if str(value['value']).isnumeric() else f"'{value['value']}'"
+            keys.append(f"{key}{value['operator']}{other_value}")
```

Comment:
Is this the only part of this PR that is actually solving the stated bug? Or is the rest relevant too?

Reply:
Yeah this line. Everything else is me cleaning up the code